packages/datagateway-download/src/downloadConfirmation/downloadConfirmDialog.component.tsx

```diff
@@ -161,15 +212,130 @@ const DownloadConfirmDialog: React.FC<DownloadConfirmDialogProps> = (
+  const broadcastError = (errorMessage: string): void => {
+    document.dispatchEvent(
+      new CustomEvent(MicroFrontendMessageId, {
+        detail: {
+          type: 'scigateway:api:notification',
+          payload: {
+            severity: 'error',
+            message: errorMessage,
+          },
+        },
+      })
+    );
+  };
+
+  // Sort access methods into a sorted array used for
+  // rendering the select dropdown list.
+  const sortMethods = React.useCallback(() => {
+    const sorted = Object.entries(statusMethods).sort(
+      ([, methodInfoA], [, methodInfoB]) => {
+        let res =
+          (methodInfoA.disabled !== undefined
+            ? methodInfoA.disabled === false
+              ? -1
+              : 0
+            : 1) -
+          (methodInfoB.disabled !== undefined
+            ? methodInfoB.disabled === false
+              ? -1
+              : 0
+            : 1);
+        return res;
+      }
+    );
+
+    // Set the sorted, selected access method
+    // and set it to have been sorted.
+    setSortedMethods(sorted);
+    setSelectedMethod(sorted[0][0]);
+    setIsSorted(true);
+  }, [statusMethods, setSortedMethods, setSelectedMethod, setIsSorted]);
+
+  React.useEffect(() => {
+    async function getStatus(): Promise<void> {
+      let statusErrors: string[] = [];
+      Promise.all(
+        Object.entries(statusMethods).map(([method]) =>
+          getDownloadTypeStatus(method, {
+            facilityName: settings.facilityName,
+            downloadApiUrl: settings.downloadApiUrl,
+          })
+        )
+      ).then(methodStatus => {
+        // Loop through all the current access methods and match that
+        // to the status information we received for each.
+        Object.entries(statusMethods).forEach(([method], index) => {
+          const status = methodStatus[index];
+          if (status) {
+            setStatusMethods(prevState => {
+              return {
+                ...prevState,
+                [method]: {
+                  ...prevState[method],
+
+                  disabled: status.disabled,
+                  message: status.message,
+                },
+              };
+            });
+          } else {
+            setStatusMethods(prevState => {
+              return {
+                ...prevState,
+                [method]: {
+                  ...prevState[method],
+
+                  disabled: undefined,
+                  message: '',
+                },
+              };
+            });
+
+            // Push the method type to display an error.
+            statusErrors.push(method);
+          }
+        });
+
+        // Broadcast errors depending on the result of the status requests.
+        if (statusErrors.length < Object.keys(statusMethods).length) {
+          for (const method of statusErrors) {
+            broadcastError(
+              `Access method ${method.toUpperCase()} is currently unavailable. If required, use an alternative method.`
```

Comment:
This error message could give a user the wrong idea. The access method might not be "currently unavailable" - what actually happened is that we could not ascertain the status of the access method. This error message should imply that the error was in fetching the status and not say whether the access method is actually enabled or disabled.